### ISSUE TITLE
Removing components from Entity in EntityListener.entityRemoved unsafe

### ISSUE THREAD

**Issue by nanodeath** [libgdx/ashley#244]
Hi,

Due to some weird little event listener/callback hell, I've discovered that if you remove a component within a EntityListener#entityRemoved callback, **it can cause that entity to be re-added into certain Families**, even though the entity was removed from the engine. Needless to say, this was very hard to track down...

I've embedded my test case below. Affects LibGDX 1.9.4.

---
```java
package name.maxaller.myparadise;

import com.badlogic.ashley.core.*;
import com.badlogic.ashley.utils.ImmutableArray;
import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class EntityListenerFamilyBug {
    @Test
    public void succeeds() {
        helper(null);
    }

    @Test
    public void alsoSucceeds() {
        helper(Component2.class);
    }

    @Test
    public void fails() {
        helper(Component1.class);
    }

    @Test
    public void alsoFails() {
        helper(Component3.class);
    }

    private void helper(final Class<? extends Component> componentToRemove) {
        Engine engine = new Engine();
        Entity entity = new Entity();
        entity.add(new Component1());
        entity.add(new Component2());
        entity.add(new Component3());
        engine.addEntity(entity);

        engine.addEntityListener(Family.all(Component1.class).get(), new EntityListener() {
            @Override
            public void entityAdded(Entity entity) { }

            @Override
            public void entityRemoved(Entity entity) {
                if (componentToRemove != null) {
                    entity.remove(componentToRemove);
                }
            }
        });

        ImmutableArray<Entity> entities = engine.getEntitiesFor(Family.all(Component2.class).get());
        assertTrue(entities.contains(entity, true));
        engine.removeEntity(entity);
        // If the test fails, it's always because the following assertion fails.
        assertFalse(entities.contains(entity, true));
    }

    private static class Component1 implements Component {
    }
    private static class Component2 implements Component {
    }
    private static class Component3 implements Component {
    }
}

```

**Comment by nanodeath** [comment]
Ah...well, I have good news! This bug only affects 1.7.0. It appears fixed in 1.7.1 and 1.7.3. (I think the LibGDX setup app initializes things to 1.7.0?) Anyway, going to resolve since it's fixed now.

**Comment by saltares** [comment]
Correct, I need to update the setup app to use the latest.
